src/server/server.service.js

```diff
@@ -15,6 +19,171 @@ module.exports = class Server {
+
+        // While there remain elements to shuffle...
+        while (currentIndex != 0) {
+            // Pick a remaining element...
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex--;
+
+            // And swap it with the current element.
+            [array[currentIndex], array[randomIndex]] = [
+                array[randomIndex], array[currentIndex]];
+        }
+
+        return array;
+    }
+
+    async fetchHeadlines(lang) {
+        if (!consts.SUPPORTED_LANGUAGES.includes(lang)) {
+            throw new Error(`Language ${lang} not supported`);
+        }
+
+        var dbConfig;
+        if (process.env.NODE_ENV === 'production') {
+            dbConfig = {
+                connectionString: process.env.DATABASE_URL,
+                ssl: {
+                    sslmode: 'require',
+                    rejectUnauthorized: false,
+                }
+            };
+        } else {
+            dbConfig = {
+                connectionString: process.env.DATABASE_URL,
+            };
+        }
+
+        const db = new Client(dbConfig);
+
+        try {
+            await db.connect();
+        } catch (err) {
+            console.error("Failed to connect to the DB: " + err);
+        }
+
+        var offset = this.headlines_offset[lang];
+        var limit = this.headlines_limit;
+
+        // fetch possible headlines
+        var q = 'SELECT headline,link ' +
+            `FROM ${lang}_headlines ` +
+            'ORDER BY id ' +
+            `LIMIT ${limit} ` +
+            `OFFSET ${offset} `;
+
+        await db.query(q)
+            .then(res => {
+                for (let row of res.rows) {
+                    debug(JSON.stringify(row));
+                }
+
+                if ((offset + limit) > this.headlines_count[lang]) {
+                    // If reached the end, go back to the beginning
+                    this.headlines_offset[lang] = 0;
+                } else {
+                    // Otherwise move the offset further
+                    this.headlines_offset[lang] = offset + limit;
+                }
+                this.headlines[lang] = this.headlines[lang].concat(this.shuffle(res.rows));
+
+                console.log(`Headlines for language ${lang} fetched`);
+            })
+            .catch(err => {
+                console.error(err);
+                throw err;
+            })
+            .finally(() => {
+                db.end();
+            });
+    }
+
+    async initHeadlines() {
+        var dbConfig;
+        if (process.env.NODE_ENV === 'production') {
+            dbConfig = {
+                connectionString: process.env.DATABASE_URL,
+                ssl: {
+                    sslmode: 'require',
+                    rejectUnauthorized: false,
+                }
+            };
+        } else {
+            dbConfig = {
+                connectionString: process.env.DATABASE_URL,
+            };
+        }
```

Comment:
How about moving setup of `dbConfig` into a dedicated function?

Reply:
Yeah, I didn't like the duplication from the beginning, but wanted to see things working faster :).

I'll fix this.

Reply:
faster or sooner, : P